### PR TITLE
Remove CAF/CCAP/certain pops/uploaded docs status columns from Applications

### PR DESCRIPTION
**Pull request by bencalegari**
- Rename ApplicationStatus to DocumentStatus (both classes and table)
- Refactored status updating into DocumentStatusRepository

[#180197141]

### CODE REVIEW

File: src/main/java/org/codeforamerica/shiba/pages/PageController.java [hunk 1]
```diff
@@ -599,9 +599,9 @@ RedirectView submitFeedback(Feedback feedback,
       Locale locale) throws IOException, InterruptedException {
     LocaleSpecificMessageSource lms = new LocaleSpecificMessageSource(locale, messageSource);
     try {
-      applicationRepository.updateStatus(applicationData.getId(), UPLOADED_DOC,
-          routingDecisionService.getRoutingDestinations(applicationData, UPLOADED_DOC),
-          IN_PROGRESS);
+      routingDecisionService.getRoutingDestinations(applicationData, UPLOADED_DOC).forEach(
+          routingDestination -> documentStatusRepository.createOrUpdate(applicationData.getId(),
+              UPLOADED_DOC, routingDestination.getName(), IN_PROGRESS));
```
Comment (andrewedstrom) [inline src/main/java/org/codeforamerica/shiba/pages/PageController.java:604]: Could we move some of this logic into DocumentStatusRepository? Probably a new method like `createOrUpdateAllForDocumentType()` but that only updates a single document type but does it for all routing destinations.

File: src/test/java/org/codeforamerica/shiba/output/MnitDocumentConsumerTest.java [hunk 2]
```diff
@@ -335,10 +349,10 @@ void updatesStatusToSendingForCafAndCcapDocuments() {
...
 
     CountyRoutingDestination routingDestination = countyMap.get(Hennepin);
-    verify(applicationRepository).updateStatus(application.getId(), CAF, routingDestination,
-        SENDING);
-    verify(applicationRepository).updateStatus(application.getId(), CCAP, routingDestination,
-        SENDING);
+    verify(documentStatusRepository).createOrUpdate(application.getId(),
+        CAF, ((RoutingDestination) routingDestination).getName(), SENDING);
+    verify(documentStatusRepository).createOrUpdate(application.getId(),
+        CCAP, ((RoutingDestination) routingDestination).getName(), SENDING);
```
Comment (andrewedstrom) [inline src/test/java/org/codeforamerica/shiba/output/MnitDocumentConsumerTest.java:355]: These type casts to `RoutingDestination` shouldn't be necessary

Comment (andrewedstrom) [reply src/test/java/org/codeforamerica/shiba/output/MnitDocumentConsumerTest.java:355]: Same applies to the whole file

**Comment by andrewedstrom** [comment]
Things we should address before merging:
- Renaming the status table from application_status to document_status will cause downtime during the deploy. If we want to do that, we should spread it across multiple releases. For now, my suggestion is to rename things in Java code like `ApplicationStatus` -> `DocumentStatusRepository` but to leave the table name as is.
- `DocumentStatusRepository.createOrUpdateAll()` uses DocumentListParser, but DocumentListParser doesn't return `UPLOADED_DOC`, so that method actually won't update the statuses of uploaded documents as it is. We should change it so it does update the statuses of uploaded docs, or we should change the name to make it more explicit.

Also left a few nitpick comments

**Comment by sree-cfa** [comment]
- Agree with @andrewedstrom that we should do the DB changes after the code changes - this is generally good practice for zero-downtime-migrations. 
- DocumentListParser is used in a few places that assumes we only want the ccap, caf, certain-pop applications. If we wanted to include uploaded_docs and xml, I would be more comfortable with discussing that a bit and making those changes in a different PR since I think it would be more complex.

File: src/main/java/db/migration/V34__BackfillApplicationStatus.java [hunk 3]
```diff
@@ -77,4 +87,24 @@ INSERT INTO application_status (application_id, document_type, routing_destinati
     }
   }
 
+  @Data
+  protected static class V34ApplicationData implements Serializable {
```
Comment (sree-cfa) [inline src/main/java/db/migration/V34__BackfillApplicationStatus.java:91]: 🙏

**Comment by andrewedstrom** [comment]
@sree-cfa Yeah I'm realizing I wasn't entirely clear.

I don't think we should change `DocumentListParser`, I think we should change `DocumentStatusRepository.createOrUpdateAll()` so that it handles `UPLOADED_DOC`s.

Definitely agree that we should not change `DocumentListParser`

**Pull request merged**